src/index.js

```diff
@@ -70,56 +82,64 @@ export default (config = {}) => {
         headers['Api-Key'] = state.apiKey
       }
 
+      const retryDelay = (attempt, error, response) => {
+        // from https://github.com/tim-kos/node-retry/blob/master/lib/retry.js
+        var random = state.transportOptions.randomize ? Math.random() + 1 : 1
+
+        var timeout = Math.round(
+          random *
+            state.transportOptions.minTimeout *
+            Math.pow(state.transportOptions.factor, attempt)
+        )
+        timeout = Math.min(timeout, state.transportOptions.maxTimeout)
+        return timeout
+      }
+
       const request = {
         method: method,
-        uri: uri,
-        qs: queryParams,
         headers: headers,
         timeout: state.transportOptions.requestTimeout,
-        resolveWithFullResponse: true
+        retries: state.transportOptions.retries,
+        retryDelay
       }
       if (options.form === true) {
         request.form = payload || {}
       } else if (options.multipart !== true) {
         request.json = true
         request.body = payload
-      } else if (typeof window !== 'undefined') {
-        request.headers['Content-Type'] = 'multipart/form-data'
+      } else {
         const formData = payload.formData || {}
-        const data = [
-          {
-            'Content-Disposition': `form-data; name="filedata"; filename="${
-              payload.filename
-            }"`,
-            body: payload.contents
-          }
-        ]
+        const requestData = new FormData()
+
+        requestData.append(payload.name, payload.contents, payload.filename)
 
         Object.keys(formData).forEach(function (meta) {
-          data.push({
-            'Content-Disposition': 'form-data; name="' + meta + '"',
-            body: JSON.stringify(formData[meta])
-          })
+          requestData.append(meta, JSON.stringify(formData[meta]))
         })
 
-        request.multipart = {
-          chunked: false,
-          data: data
-        }
-      } else {
-        const formData = payload.formData || {}
-
-        formData[payload.name] = {
-          value: payload.contents,
-          options: {
-            filename: payload.filename
-          }
-        }
+        request.body = requestData
+      }
 
-        request.formData = formData
+      if (request.json && request.body && typeof request.body === 'object') {
+        request.body = JSON.stringify(request.body)
       }
 
-      return transport(request, state.transportOptions)
+      const t = transport(uri, request)
+      // currently in the tests, we don't have then...
```

Comment:
hm, weird. Does it depend on node version?

Reply:
No. We "overwrite" `transport` in the test with testdouble to check if we call it with the right parameters. So that isn't a Promise then and doesn't have `then`. I'm sure there would be a way to have that, but couldn't figure it out in a decent amount of time.

Reply:
ah right.. :/ sucks. But yeah, ok for now. The whole testdouble thing is a bit questonable, but anyway ;)